pkg/ebpf/events/derived/derived.go

```diff
@@ -0,0 +1,29 @@
+package derived
+
+import "github.com/aquasecurity/tracee/types/trace"
+
+// DeriveFn is a function prototype for a function that receives an event as
+// argument and may produce a new event if relevant.
+// It returns list of derived events, which might be empty depending on successful derivation,
+// a bool indicating if an event was derived, and an error if one occurred.
+type DeriveFn func(trace.Event) ([]trace.Event, bool, error)
```

Comment:
Why would we return a list of derived events? I assume it's for the ability to derive multiple events from one, but currently our mechanism for deriving one events into multiple is declaring multiple derivable events in the derivation table (events_derived.go:18).

So we could drop the map to map idea and use a list of derived events instead, as you do here, but we don't currently have an event that needs multiple derivation, as far as I can tell from the code, and secondly, why would it be better than the solution we currently have (performance, readability, etc.)?

I do see how we could use this to return multiple events of the same derivation, but again, do we actually need such a feature (Is it related to #1823)?

Reply:
The current structure enable me to create a single derived event from a single event.
I can create multiple events from multiple events, but I can't create for a single `cgroup_makedir` event more than 1 `container_create` event. In my case, I want to create for a single event of SO loading multiple events of collisions with previous loaded SOs. The current structure just won't allow it, and for no good reason.

Reply:
Alright, so it is related to #1823, I can go with that, but could you also document at the comment that a DeriveFn should return events of a singular type?

Reply:
Also, I think we could use this to reduce the api from ([]trace.Event, bool, err) to just ([]trace.Event, []error). I initially included the bool to indicate that while no error occurred in the derivation, the event was irrelevant and empty, so it shouldn't be propagated.
But now we could just check for `len(derivatives) != 0`, or even just loop over the slice without such a check (if no error occured).
However, because we can derive multiple events at once now, this should now return a `[]error`, since we can have multiple derivation errors.